jumpscale/sals/process/__init__.py

```diff
@@ -84,595 +141,833 @@ def execute(
-    """Kill all processes for a given command
+def kill_process_by_name(process_name, sig=signal.SIGTERM, match_predicate=None, timeout=5, sure_kill=False):
+    """Kill all processes that match 'process_name'.
 
-    Arguments:
-        name {str} -- Name of the command that started the process(s)
+    Args:
+        process_name (str): The target process name.
+        sig (signal, optional): See signal module constants. Defaults to signal.SIGKILL
+        match_predicate (callable, optional): Function that does matching between\
+            found processes and the targeted process, the function should accept\
+            two arguments and return a boolean. Defaults to None.
+        timeout (int, optional): How long to wait for a process to terminate (seconds) before raise exception\
+            or, if sure_kill=True, send a SIGKILL. Defaults to 5.
+        sure_kill (bool, optional): Whether to fallback to SIGKILL if the timeout exceeded for the terminate operation or not. Defaults to False.
 
-    Keyword Arguments:
-        sig {bool} -- os signal to send to the process(s) (default: {signal.SIGTERM.value})
-        match_predicate {callable} -- function that does matching between
-            found processes and the targested process, the function should accept
-            two arguments and return a boolean (default: {None})
+    Returns:
+        None or list of int: represents the IDs of the processes remaning alive.
     """
+    pids = get_pids(process_name, match_predicate=match_predicate)
+    failed_processes = []
+    for pid in pids:
+        try:
+            kill(pid, sig, timeout=timeout, sure_kill=sure_kill)
+        except (j.exceptions.Runtime, j.exceptions.Permission):
+            failed_processes.append(pid)
+    return failed_processes or None  # return None if the failed list is empty
+
+
+def kill_all_pids(pids, sig=signal.SIGTERM, timeout=5, sure_kill=False):
+    """Kill all processes with given pids.
 
-    pids = get_pids(name, match_predicate=match_predicate)
+    Args:
+        pids (list of int): The target processes IDs.
+        sig (signal, optional): See signal module constants. Defaults to signal.SIGKILL.
+        timeout (int, optional): How long to wait for a process to terminate (seconds) before raise exception\
+            or, if sure_kill=True, send a SIGKILL. Defaults to 5.
+        sure_kill (bool, optional): Whether to fallback to SIGKILL if the timeout exceeded for the terminate operation or not. Defaults to False.
+
+
+    Returns:
+        None or list of int: represents the IDs of the processes remaning alive.
+    """
+    failed_processes = []
     for pid in pids:
-        kill(pid, sig)
+        try:
+            kill(pid, sig, timeout=timeout, sure_kill=sure_kill)
+        except (j.exceptions.Runtime, j.exceptions.Permission):
+            failed_processes.append(pid)
+    return failed_processes or None  # return None if the failed list is empty
```

Comment:
The same suggestion for returning empty list instead.

Reply:
The same as a previous replay on similar comments :D